src/raftstore/store/peer.rs

```diff
@@ -455,8 +459,12 @@ impl Peer {
                   ready.hs.is_some());
 
         if is_applying {
-            // remove hard state so raft won't change the apply index.
+            // Remove the hard state so Raft won't change the apply index.
             ready.hs.take();
+        } else if apply_index != 0 {
+            if let Some(ref mut hs) = ready.hs {
+                hs.set_commit(apply_index);
```

Comment:
if committed entries have not been applied completely, the peer should be append to pending_raft_group again.